x-pack/plugin/security/qa/security-basic/src/javaRestTest/java/org/elasticsearch/xpack/security/ManageRolesPrivilegeIT.java

```diff
@@ -0,0 +1,211 @@
+                        new RoleDescriptor.IndicesPrivileges[] {
+                            RoleDescriptor.IndicesPrivileges.builder().indices("test-allowed-suffix").privileges("manage").build() },
+                        new RoleDescriptor.ApplicationResourcePrivileges[0],
+                        new ConfigurableClusterPrivilege[0],
+                        new String[0],
+                        Map.of(),
+                        Map.of()
+                    )
+                )
+            );
+            assertThat(
+                responseException.getMessage(),
+                containsString("this action is granted by the cluster privileges [manage_security,all]")
+            );
+        }
+    }
+
+    public void testManageSecurityNullifiesManageRoles() throws Exception {
+        createManageRolesRole("manage-roles-no-manage-security", new String[0], Set.of("allowed"));
+        createManageRolesRole("manage-roles-manage-security", new String[] { "manage_security" }, Set.of("allowed"));
+
+        createUser("test-user-no-manage-security", Set.of("manage-roles-no-manage-security"));
+        createUser("test-user-manage-security", Set.of("manage-roles-manage-security"));
+
+        String authHeaderNoManageSecurity = basicAuthHeaderValue("test-user-no-manage-security", TEST_PASSWORD);
+        String authHeaderManageSecurity = basicAuthHeaderValue("test-user-manage-security", TEST_PASSWORD);
+
+        createRole(
+            authHeaderNoManageSecurity,
+            new RoleDescriptor(
+                "test-role-allowed-by-manage-roles",
+                new String[0],
+                new RoleDescriptor.IndicesPrivileges[] {
+                    RoleDescriptor.IndicesPrivileges.builder().indices("allowed").privileges("read").build() },
+                new RoleDescriptor.ApplicationResourcePrivileges[0],
+                new ConfigurableClusterPrivilege[0],
+                new String[0],
+                Map.of(),
+                Map.of()
+            )
+        );
+
+        ResponseException responseException = assertThrows(
+            ResponseException.class,
+            () -> createRole(
+                authHeaderNoManageSecurity,
+                new RoleDescriptor(
+                    "test-role-not-allowed-by-manage-roles",
+                    new String[0],
+                    new RoleDescriptor.IndicesPrivileges[] {
+                        RoleDescriptor.IndicesPrivileges.builder().indices("not-allowed").privileges("read").build() },
+                    new RoleDescriptor.ApplicationResourcePrivileges[0],
+                    new ConfigurableClusterPrivilege[0],
+                    new String[0],
+                    Map.of(),
+                    Map.of()
+                )
+            )
+        );
+
+        assertThat(
+            responseException.getMessage(),
+            // TODO Should the new global role/manage privilege be listed here? Probably not because it's not documented
```

Comment:
I think it would require some refactoring to achieve this. Not sure if it's worth it at this point since this privilege is used only for connectors. What do you think @slobodanadamovic ?

Reply:
I agree. Let's keep it for now.